### PR TITLE
Add max_interleaving DPC++ FPGA tutorial

### PR DESCRIPTION
**Pull request by citian-intel**
Add max_interleaving DPC++FPGA tutorial

# Description

This tutorial demonstrates how to apply the max_interleaving attribute to loops and how it affects loop resource use.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [X] Command Line

### CODE REVIEW

**Comment by citian-intel** [comment]
@akertesz @AlbyB @pmpeter1 @tomlenth please review.

**Comment by tomlenth** [comment]
This looks good to me.  thank you.

**Comment by akertesz** [comment]
Your code is written for DPC++ compiler beta10 version. However, we require backwards compatibility with the beta09 compiler until the beta10 version is formally released.

Please add the following code for backwards compatability:
```
// Header locations and some DPC++ extensions changed between beta09 and beta10
// Temporarily modify the code sample to accept either version
#define BETA09 20200827
#if __SYCL_COMPILER_VERSION <= BETA09
  #include <CL/sycl/intel/fpga_extensions.hpp>
  namespace INTEL = sycl::intel;  // Namespace alias for backward compatibility
#else
  #include <CL/sycl/INTEL/fpga_extensions.hpp>
#endif
```

You can refer to https://github.com/oneapi-src/oneAPI-samples/pull/167 to see an example of this.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]
Missing header file location comment above "dpc_common.hpp"

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp [hunk 1]
```diff
@@ -0,0 +1,202 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <array>
+#include <iomanip>
+#include <iostream>
+#include "dpc_common.hpp"
```
Comment (JoeOster) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp:10]: Please add the comments below for for the include of dpc_common.hp. it should look like this:

// dpc_common.hpp can be found in the dev-utilities include folder.
// e.g., $ONEAPI_ROOT/dev-utilities/<version>/include/dpc_common.hpp
#include "dpc_common.hpp"

as seen in https://github.com/oneapi-src/oneAPI-samples/blob/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot/src/main.cpp

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+// Header locations and some DPC++ extensions changed between beta09 and beta10
+// Temporarily modify the code sample to accept either version
+#include <CL/sycl.hpp>
```
Comment (JoeOster) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp:14]: includes should be in alphabetical order

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp (hunk 1 above)
Comment (citian-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp:10]: Thanks for the suggestions, I have updated the source code accordingly.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
```
Comment (JoeOster) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp]: In your repo, line 7 is repeated on line 17
#include <CL/sycl.hpp>

**Review by JoeOster — CHANGES_REQUESTED** [review]

**Review by pmpeter1 — CHANGES_REQUESTED** [review]
You should not be including <CL/sycl.hpp> twice.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp (hunk 3 above)
Comment (citian-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp]: addressed

**Review by pmpeter1 — APPROVED** [review]

File: DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp (hunk 2 above)
Comment (citian-intel) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/max_interleaving/src/max_interleaving.cpp]: address as well

**Comment by akertesz** [comment]
Windows CI failed -- looks like a basic copy-paste error. Please update your ninja file to reference the correct files!

**Review by akertesz — APPROVED** [review]

**Comment by akertesz** [comment]
Approved to re-launch the CI test

**Comment by citian-intel** [comment]
I have tested the windows flow locally, adding an explicit scope to the sqrt function fix the failures on windows.

**Review by akertesz — APPROVED** [review]
Let's retrigger the CI test to see whether the sqrt issue is resolved.

**Review by JoeOster — APPROVED** [review]

**Comment by akertesz** [comment]
Paul's comment is addressed, Tom is satisfied with the README, and CI is passing -- I think we're good to go!

**Pull request merged**